.github/workflows/test-api.yml

```diff
@@ -12,8 +12,23 @@ on:
       - 'libs/dal/**'
       - 'libs/testing/**'
       - 'libs/shared/**'
-
+  workflow_dispatch:
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   test_api:
     uses: ./.github/workflows/reusable-api-e2e.yml
+
+  build_docker:
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    permissions:
+      contents: read
+      packages: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-project
+      - uses: ./.github/actions/docker/build-api
```

Comment:
Amazing 💯

Reply:
Thought about something, what about creating a separate action that will be triggered in order to build the app (API for example), this way we could remove the complexity of the if statement, and it could run independently and be parallelized

@scopsy what do you think?

Reply:
I'm not sure i follow 100% percent. Do you mind explaining a bit more or giving some concrete example?

Reply:
Sure thing, after looking at it again I think we can create an additional flow that will be triggered for example 'api-image-build' so we can separate the so flow and parallelize the two flows as well. it is not necessary at all just thought it will help during the testing time.